odml/base.py

```diff
@@ -136,11 +136,17 @@ def __contains__(self, key):
             if (hasattr(obj, "name") and obj.name == key) or key == obj:
                 return True
 
-    def append(self, obj):
-        if obj.name in self:
-            raise KeyError("Object with the same name already exists! " + str(obj))
-        else:
-            super(SmartList, self).append(obj)
+    def append(self, *obj_tuple):
+        ll = len(obj_tuple)
+        for i in list(range(0,ll)):
```

Comment:
I am sorry to be picky but this construct is a bit verbose. 
What about:
````
for i in range(len(obj_tuple)):
    ....
````

or even:
 ````
for obj in obj_tuple:
     if obj.name in self:
          ....
````
This would save a few operations  and should lead to better performance